src/components/TextEntry.vue

```diff
@@ -0,0 +1,90 @@
+<template>
+  <div class="flex flex-col justify-center">
+    <p
+      class="w-1/2 text-xl lg:text-2xl xl:text-3xl mx-auto font-bold md:w-3/4 text-center"
+    >
+      {{ label }}
+    </p>
+    <div class="flex flex-row justify-center">
+      <input
+        v-model="text"
+        type="text"
+        inputmode="text"
+        :placeholder="placeholder"
+        :required="isRequired"
+        class="border-2 rounded-md p-4 mx-auto border-gray-500 focus:border-gray-800 focus:outline-none mt-2"
+        :class="selectInputBoxClasses()"
+        @keypress="isValidTextEntry($event)"
+        @input="updateTextEntry($event)"
+        ondrop="return false"
+        onpaste="return false"
+      />
+    </div>
+    <span
+      v-if="isInvalidTextEntryMessageShown"
+      class="mx-auto text-red-700 text-base mb-1"
+      >{{ invalidTextEntryMessage }}</span
+    >
+  </div>
+</template>
+<script>
+import { validationTypeToFunctionMap } from "@/services/basicValidationMapping.js";
+
+export default {
+  name: "TextEntry",
+  emits: ["valid-entry"],
+  props: {
+    label: {
+      type: String,
+      default: "",
+    },
+    placeholder: {
+      type: String,
+      defult: "",
+    },
+    isRequired: {
+      type: Boolean,
+      default: false,
+    },
+    key: {
+      type: String,
+      default: "",
+    },
+  },
+  data() {
+    return {
+      text: "",
+      invalidTextEntryMessage: "",
+    };
+  },
+  computed: {
+    /** Returns if message for an invalid text entry needs to be shown */
+    isInvalidTextEntryMessageShown() {
+      return this.invalidTextEntryMessage != null;
+    },
+  },
+  methods: {
+    /** Determines how the input box should look.
+     */
+    selectInputBoxClasses() {
+      return [
+        {
+          "border-red-600 focus:border-red-600": this.invalidTextEntryMessage,
+        },
+      ];
+    },
+    /** Updates text based on user entry */
+    updateTextEntry(event) {
+      if (event.target.value.length == 0) {
+        this.invalidTextEntryMessage = "";
```

Comment:
we can have a maxTextLength here na just like number and phonenumber

Reply:
How would you decide what the upper limit would be? Text does not limits. In fact, it can a min limit.